### ISSUE TITLE
./monorail_rack isn't launching example VM

### ISSUE THREAD

**Issue by heckj** [RackHD/RackHD#4]
@DavidjohnBlodgett was following along with the example flow and conversation from chat, and after the dev VM was built and everything shoved in the right places, I hit an error when the script attempted to fire up a sample PXE client VM:

```
==> dev: VirtualBox VM is already running.
deploying pxe: 1
VBoxManage: error: Machine settings file '/Users/heckj/VirtualBox VMs/pxe-1/pxe-1.vbox' already exists
VBoxManage: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MachineWrap, interface IMachine, callee nsISupports
VBoxManage: error: Context: "CreateMachine(bstrSettingsFile.raw(), bstrName.raw(), ComSafeArrayAsInParam(groups), bstrOsTypeId.raw(), createFlags.raw(), machine.asOutParam())" at line 272 of file VBoxManageMisc.cpp
0%...
Progress state: VBOX_E_FILE_ERROR
VBoxManage: error: Failed to create medium
VBoxManage: error: Could not create the medium storage unit '/Users/heckj/src/rackhd/rackhd/example/bin/pxe-1.vdi'.
VBoxManage: error: VDI: cannot create image '/Users/heckj/src/rackhd/rackhd/example/bin/pxe-1.vdi' (VERR_ALREADY_EXISTS)
VBoxManage: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MediumWrap, interface IMedium
VBoxManage: error: Context: "RTEXITCODE handleCreateMedium(HandlerArg*)" at line 448 of file VBoxManageDisk.cpp
VBoxManage: error: Storage controller named 'SATA Controller' already exists
VBoxManage: error: Details: code VBOX_E_OBJECT_IN_USE (0x80bb000c), component SessionMachine, interface IMachine, callee nsISupports
VBoxManage: error: Context: "AddStorageController(Bstr(pszCtl).raw(), StorageBus_SATA, ctl.asOutParam())" at line 1061 of file VBoxManageStorageController.cpp
```

^^ you seen this? When I kick open VirtualBox, the VM is created and powered off - so I'm wondering if this just isn't happen when I re-run even after the VM was created.

**Comment by DavidjohnBlodgett** [comment]
So I have seen errors like this in the past, but for a different reason I believe... So from what I read, I am going to go out on a limb and think, your assuming the monorail_rack command will automatically start the pxe clients. However, it does not so once the script has created them, you will want to open Virtual Box and manually boot each of the pxe clients in the order you wish.

**Comment by benbp** [comment]
I think what he's also saying is if the clients are already created, we shouldn't print an error, we should just ignore them.

It might also make sense, as a feature, to improve the monorail_rack with flags like:

```
monorail_rack create 2
monorail_rack boot 2
```

(I have already written the boot/shutdown/reboot logic on my own for my .bashrc and you're welcome to steal it).

**Comment by DavidjohnBlodgett** [comment]
ohhh, sorry I did miss read that... Yes I agree the script should not give you those ugly errors. It was only acceptable when it was an internal tool. I'll plan to fix this, and @benbp I had always wanted to add a command line to override what was read in from the config, but was focused on delivering on the first milestone, now seems appropriate to put it back in so will plan to update this aswell.

**Comment by heckj** [comment]
yep, that's what happened - and what Ben said applies too. I think what really confused me is that I didn't know the VM was even made. Is there a command extension to have the PXE vm's show up with the graphic interface on by default, instead of the vagrant standard of hidden? 

I didn't realize it was there because the Vagrant file only had a single definition with in, and you spin up the PXE vm's with the virtual box server commands directly.

**Comment by DavidjohnBlodgett** [comment]
So I was circling back on this and noticed that the above error was referencing this path 
/Users/heckj/VirtualBox VMs/pxe-1/pxe-1.vbox

I have not been able to pin down when this actually happens... but once in a while I would find a pxe client seem to be disembodied from Virtual box... basically a folder would be created in a path like shown above but no disk could be created to bring it back to life... I found a workaround in this unique case,

1) verify you see no pxe clients in virtualbox GUI.
2) verify there is no pxe-1.vdi file in the example/bin directory.
3) then delete  the problem directory,
     rm -rf /Users/heckj/VirtualBox VMs/pxe-1/

That effectively resets the Virtual Box environment and the errors go away.

**Comment by heckj** [comment]
resolved while working on #14
